### PR TITLE
Add `pocket-knife` icon

### PR DESCRIPTION
**Pull request by danielbayley**
Alternatives:

Untitled

### CODE REVIEW

File: icons/pocket-knife.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  width="24"
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <rect width="6" height="20" x="9" y="2" rx="3" />
```
Comment (jguddas) [inline icons/pocket-knife.svg:12]: The body needs to be bigger as to not violate the 2px rule
```suggestion
  <rect x="8" y="2" width="8" height="20" ry="4" />
```

File: icons/pocket-knife.svg [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <path d="M12 5h0" />
+  <path d="M12 19h0" />
```
Comment (jguddas) [inline icons/pocket-knife.svg:14]:
```suggestion
  <path d="M12 6h.01" />
  <path d="M12 18h.01" />
```

**Review by jguddas — CHANGES_REQUESTED** [review]

**Comment by jguddas** [comment]
Very cute, not sure about the use case but I'm a fan :+1:

**Comment by karsa-mistmere** [comment]
Really not sure about this, the original proposal definitely violated the 2px gap guideline, on the other hand, this new one doesn't quite look like a pocket knife any more. 🤷‍♂️

**Comment by karsa-mistmere** [comment]
Not convinced about this, but maybe something similar?
image

As usually, our [required icon request input](https://github.com/lucide-icons/lucide/issues/new?assignees=&labels=%F0%9F%99%8C+icon+request&template=icon_request.md) would be really helpful here (use cases & similar icons).

**Comment by karsa-mistmere** [comment]
I think let's go for just `pocket-knife-2` out of the two, it's more legible and fits the "multi tool" use case better.

File: icons/pocket-knife.svg [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M18 6h0" />
+  <path d="M6 18h0" />
+  <path d="M8.9 20.8c-1.6 1.6-4.1 1.6-5.7 0-1.6-1.6-1.6-4.1 0-5.7L15.1 3.2c1.6-1.6 4.1-1.6 5.7 0 1.6 1.6 1.6 4.1 0 5.7L8.9 20.8z" />
+  <polyline points="4.6 13.6 2 11 4 9 2 7 4 5 2 3 3 2" />
+  <path d="M18 12v10h0c2.2 0 4-1.8 4-4V6" />
```
Comment (karsa-mistmere) [inline icons/pocket-knife.svg:16]: Some optimization + fixes for mismatched paths.
```suggestion
  <path d="M18 6h.01" />
  <path d="M6 18h.01" />
  <path d="M20.83 8.83a4 4 0 0 0-5.66-5.66l-12 12a4 4 0 1 0 5.66 5.66Z" />
  <path d="M3 2 2 3l2 2-2 2 2 2-2 2 2.67 2.67" />
  <path d="M18 11.66V22a4 4 0 0 0 4-4V6" />
```

**Review by karsa-mistmere — APPROVED** [review]

**Comment by danielbayley** [comment]
> In @karsa-mistmere s version the corkscrew has more swoosh, I like swooshieness.

@jguddas Swooshy enough…?

**Review by ericfennis — APPROVED** [review]

**Pull request merged**